src/paulscode/android/mupen64plusae/input/provider/AbstractProvider.java

```diff
@@ -206,6 +244,34 @@ public static String getHardwareName( int id )
     }
     
     /**
+     * Gets the unique name of the device.
+     * 
+     * @param id The unique hardware identifier.
+     * 
+     * @return The unique name of the device, or null if hardware not found.
+     */
+
+    @SuppressLint("NewApi")
+	public static String getUniqueName( int id )
+    {
+        if( id > HARDWARE_ID_MOGA_OFFSET && id <= HARDWARE_ID_MOGA_MAX )
+        {
+            return mogaPrefix + ( id - HARDWARE_ID_MOGA_OFFSET );
```

Comment:
This seems correct to me, but I'm unable to test to make sure this works correctly with Moga devices. @littleguy77 Do we have any testers that have moga controllers.

Worst comes to worst, if it doesn't work right, it should still behave like before, they would have to manually re-map the moga controller again to a player.